### ISSUE TITLE
Diagnostic messages for the deprecated List constructor are inconsistent

### ISSUE THREAD

**Issue by jamesderlin** [dart-lang/sdk#47564]
Attempting to use the deprecated `List` constructor results in multiple diagnostic messages:

> ```
>    info • foo.dart:2:9 • 'List' is deprecated and shouldn't be used. Use a
>           list literal, [], or the List.filled constructor instead. Try
>           replacing the use of the deprecated member with the replacement. •
>           deprecated_member_use
> ```

> ```
>   error • foo.dart:2:9 • The default 'List' constructor isn't available when
>           null safety is enabled. Try using a list literal, 'List.filled' or
>           'List.generate'. • default_list_constructor
> ```

1. The `deprecated_member_use` message's use of the comma is confusing because it looks like it's presenting a list of three options, but `, [],` is meant to be a parenthetical element describing "list literal".  I think it'd be better to just remove `, [],` entirely or reword it to "Use [] to create a list literal or use the List.filled constructor instead".
2. Unlike the `deprecated_member_use` message, the `default_list_constructor` message suggests `List.generate`.  The messages should be consistent.
3. If null-safety is enabled, ideally the `deprecated_member_use` message wouldn't be shown at all, although unless there's some way to make `@Deprecated` version-specific, I suspect that there isn't anything that could easily be done about that.

I've observed this with Dart SDK 2.14.4.

**Comment by srawlins** [comment]
I imagine this is stale, as the element has actually been removed now.

**Comment by lrhn** [comment]
ACK. Fixed by targeted procrastination.
